console/frontend/src/main/java/nl/nn/adapterframework/console/ConsoleFrontend.java

```diff
@@ -0,0 +1,142 @@
+/*
+   Copyright 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.console;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.context.annotation.Scope;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ResourceUtils;
+
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import nl.nn.adapterframework.lifecycle.DynamicRegistration;
+
+/**
+ * Contains the component annotation so it will be picked up by the core (SpringEnvironmentContext.xml).
+ * When using Spring boot (and the FrankConsoleContext.xml), where classpath scanning is enabled, 
+ * it will directly try to configure this servlet without any configuration.
+ * Hence it should use the RegisterServletEndpoints which uses a ServletConfiguration to initialize the Servlet.
+ * 
+ * @author Niels Meijer
+ */
+@Log4j2
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_SINGLETON)
+public class ConsoleFrontend extends HttpServlet implements DynamicRegistration.Servlet, EnvironmentAware, InitializingBean {
+
+	private static final long serialVersionUID = 1L;
+	private static final String WELCOME_FILE = "index.html";
+	private static final String DEFAULT_CONSOLE_PATH = "classpath:/console";
+
+	private transient String frontendPath = null;
```

Comment:
Is it a good idea to make this transient?

I don't know under what circumstances the instance could be serialised and deserialised, but after that we will lose both the environment and the frontEndPath.

I have a feeling either could be risky and unexpected.

Reply:
You're right, the frontendPath should be saved, the environment doesn't 😃